test/test_transforms.py

```diff
@@ -1869,7 +1878,8 @@ def test_random_rotation():
         transforms.RandomRotation([-0.7, 0, 0.7])
 
     t = transforms.RandomRotation(0, fill=None)
-    assert t.fill == 0
+    # TODO: BC-break - do we care?
+    assert t.fill == defaultdict()
```

Comment:
@vfdev-5 and I discussed offline and our answer is no. As explained in #6517, `fill=None` and `fill=0` is not the same. However, some v1 transforms (the three highlighted here) simply coerce `None` into `0`:

https://github.com/pytorch/vision/blob/55d3ba621fc49cc7352167e64bc0cb02bffef9a7/torchvision/transforms/transforms.py#L1338-L1339

AFAIK this behavior is undocumented. In v2 we pass through `None` values:

https://github.com/pytorch/vision/blob/55d3ba621fc49cc7352167e64bc0cb02bffef9a7/torchvision/transforms/v2/_utils.py#L56-L61

Note that in both cases, `fill=None` is *not* the default value

https://github.com/pytorch/vision/blob/55d3ba621fc49cc7352167e64bc0cb02bffef9a7/torchvision/transforms/transforms.py#L1321

https://github.com/pytorch/vision/blob/55d3ba621fc49cc7352167e64bc0cb02bffef9a7/torchvision/transforms/v2/_geometry.py#L333

So v1 is silently ignoring user input, while v2 doesn't.

Reply:
Right now, this assertion should be
```
assert t.fill is None
```